js/layout-javascript/agenda-code.js

```diff
@@ -886,6 +886,12 @@ DynamicList.prototype.initialize = function() {
     .then(function(records) {
       _this.listItems = _this.getPermissions(records);
 
+      _this.listItems.map(function(item) {
```

Comment:
In the few lines above this, we have `_this.Utils.Records.prepareData()` that looks through all the records to prepare them.

Then, you've got, in this section, code that loops through all the records **again** and then looping through all the properties within them to reassign them. This is highly inefficient.

If you want to pre-process something, do it within the `_this.Utils.Records.prepareData()` function -- though I don't see why that would be necessary for this feature.